### DISCUSSION TITLE
Can't make a KTX texture array

### DISCUSSION THREAD

**Discussion by tuket** [KhronosGroup/KTX-Software#452] [Q&A] [answered]
Hi,
I have a bunch of PNG files and I want to put all them in a single KTX as a texture array.
I've tried something like this:
```
toktx.exe --bcmp --automipmap --layers 5 "out.ktx" "0.png" "1.png" "3.png" "4.png"
```

But the out.ktx doesn't seem to have multile layers (ktxinfo.exe report 0 layers).

Is this a bug or I'm doing something wrong?
I have tried many cmd combinations: removing --automipmaps, using --t2, changing --bcmp to --uastc...
None of them worked.

**Comment by tuket** [comment, marked as answer]
I've been debugging it.

![image](https://user-images.githubusercontent.com/4165974/126747945-72fff603-d891-40e3-860b-8a30172ecfd5.png)

Looks like it's not even supported :(
I think the tool should at least warn you, I wasted a lot of time with this.

**Reply by MarkCallow** [reply]
Sorry you wasted a lot of time. Most of the work for supporting arrays has been done but TODO lines were overlooked. I'll get to it soon.

**Reply by tuket** [reply]
Thanks, I think I got it working. I have made a pull request, any thoughts? https://github.com/KhronosGroup/KTX-Software/pull/459

**Reply by MarkCallow** [reply]
Arrays and 3d textures were fixed  by PR #468 based partly on your PR @tuket. Thanks for your help.

**Comment by tuket** [comment]
Besides that, I was getting a crash while debugging.
I think this line of code is missing:
```cpp
// Input file order is layer, faceSlice, level. This seems easier for
// a human to manage than the order in a KTX file. It keeps the
// base level images and their mip levels together.
level++;
levelWidth >>= 1;
levelHeight >>= 1;
levelDepth >>= 1;
if (level == levelCount) {
    faceSlice++;
    level = 0;
    levelWidth = createInfo.baseWidth;
    levelHeight = createInfo.baseHeight;
    levelDepth = createInfo.baseDepth; // ADDED this line
    if (faceSlice == (options.cubemap ? 6 : levelDepth)) {
        faceSlice = 0;
        layer++;
        if (layer == options.layers) {
            // We're done.
            break;
        }
    }
}
```

**Comment by bzm3r** [comment]
I ran into this issue today, because I downloaded the release tagged `latest` rather than one of the pre-release candidates...

**Comment by MarkCallow** [comment]
@bzm3r please use a pre-release version. Top of trunk (master) is ready for release. The holdup is getting a working signing certificate for signing the Windows installers. Any day now...
